sdk/core/azure-core/src/main/java/com/azure/android/core/util/logging/ClientLogger.java

```diff
@@ -0,0 +1,289 @@
+    public void error(String format, Object... args) {
+        log(ERROR_LEVEL, format, args);
+    }
+
+    /*
+     * This method logs the formattable message if the {@code logLevel} is enabled
+     *
+     * @param logLevel The log level at which this message should be logged
+     * @param format The formattable message to log
+     * @param args Arguments for the message, if an exception is being logged last argument is the throwable.
+     */
+    private void log(int logLevel, String format, Object... args) {
+        if (canLogAtLevel(logLevel)) {
+            performLogging(logLevel, false, format, args);
+        }
+    }
+
+    /**
+     * Logs the {@link RuntimeException} at the warning level and returns it to be thrown.
+     *
+     * @param runtimeException RuntimeException to be logged and returned.
+     * @return The passed {@code RuntimeException}.
+     * @throws NullPointerException If {@code runtimeException} is {@code null}.
+     */
+    public RuntimeException logExceptionAsWarning(RuntimeException runtimeException) {
+        return logException(runtimeException, WARNING_LEVEL);
+    }
+
+    /**
+     * Logs the {@link RuntimeException} at the error level and returns it to be thrown.
+     *
+     * @param runtimeException RuntimeException to be logged and returned.
+     * @return The passed {@code RuntimeException}.
+     * @throws NullPointerException If {@code runtimeException} is {@code null}.
+     */
+    public RuntimeException logExceptionAsError(RuntimeException runtimeException) {
+        return logException(runtimeException, ERROR_LEVEL);
+    }
+
+    private RuntimeException logException(RuntimeException runtimeException, int logLevel) {
+        Objects.requireNonNull(runtimeException, "'runtimeException' cannot be null.");
+
+        if (canLogAtLevel(logLevel)) {
+            performLogging(logLevel, true, runtimeException.getMessage(), runtimeException);
+        }
+
+        return runtimeException;
+    }
+
+    /*
+     * Performs the logging.
+     *
+     * @param format formattable message.
+     * @param args Arguments for the message, if an exception is being logged last argument is the throwable.
+     */
+    private void performLogging(int logLevel, boolean isExceptionLogging, String format,
+                                Object... args) {
+        // If the logging level is less granular than verbose remove the potential throwable from the args.
+        String throwableMessage = "";
+        if (doesArgsHaveThrowable(args)) {
+            // If we are logging an exception the format string is already the exception message, don't append it.
+            if (!isExceptionLogging) {
+                Object throwable = args[args.length - 1];
+
+                // This is true from before but is needed to appease SpotBugs.
+                if (throwable instanceof Throwable) {
+                    throwableMessage = ((Throwable) throwable).getMessage();
+                }
+            }
```

Comment:
In the Java SDK there's a [piece of code](https://github.com/Azure/azure-sdk-for-java/blob/a2fa92b55d4c82000ccc64c135efbe1a9dd45746/sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java#L201) that removes the throwable from he args array at the end of this block. I think it would be valuable to keep it.